pkg/issuer/acme/dns/digitalocean/digitalocean.go

```diff
@@ -53,8 +53,10 @@ func NewDNSProviderCredentials(token string, dns01Nameservers []string, userAgen
 	unusedCtx := context.Background() // context is not actually used
 	c := oauth2.NewClient(unusedCtx, oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token}))
 
-	clientOpts := []godo.ClientOpt{godo.SetUserAgent(userAgent)}
-	client, err := godo.New(c, clientOpts...)
+	client, err := godo.New(c,
+		godo.SetUserAgent(userAgent),
+		godo.WithRetryAndBackoffs(godo.RetryConfig{RetryMax: 3}),
```

Comment:
I wonder if we shouldn't to the following:

```suggestion
		godo.WithRetryAndBackoffs(godo.RetryConfig{RetryMax: math.MaxInt}),
```

Currently, after 3 attempts, cert-manager will re-queue the failing request anyway. Doing this, we ensure that exponential backoff works properly.

WDYT?